### PR TITLE
dt.weekday_name - removal of function

### LINKED ISSUE #8610: dt.weekday_name - attribute removed from Pandas v1.0.0
### What happened?

Calling the `weekday_name` such as:
`da.dt.weekday_name`
raises an exception.

Looking into it more deeply, i.e. xarray source it can be seen the values are converted to a Pandas Series and the attribute called directly from that series.

This attribute was [removed from Pandas in v1.0.0](https://pandas.pydata.org/docs/whatsnew/v1.0.0.html)
> Removed Timestamp.weekday_name, DatetimeIndex.weekday_name, and Series.dt.weekday_name ([GH 18164](https://github.com/pandas-dev/pandas/issues/18164))

It was replaced with the method [day_name()](https://pandas.pydata.org/docs/reference/api/pandas.Series.dt.day_name.html#pandas.Series.dt.day_name) - so the code would need to consider this case separately.

### What did you expect to happen?

Return the weekday_name.

### Minimal Complete Verifiable Example

```Python
import pandas as pd
import xarray as xr
da = xr.DataArray(pd.date_range("2024-01-15", periods=2))
day_name = da.dt.weekday_name
```

### MVCE confirmation

- [x] Minimal example — the example is as focused as reasonably possible to demonstrate the underlying issue in xarray.
- [x] Complete example — the example is self-contained, including all data and the text of any traceback.
- [x] Verifiable example — the example copy & pastes into an IPython prompt or [Binder notebook](https://mybinder.org/v2/gh/pydata/xarray/main?urlpath=lab/tree/doc/examples/blank_template.ipynb), returning the result.
- [x] New issue — a search of GitHub Issues suggests this is not a duplicate.
- [x] Recent environment — the issue occurs with the latest version of xarray and its dependencies.

### Relevant log output

```Python
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/sam/Documents/GitHub/filebasedbim/venv/lib/python3.12/site-packages/xarray/core/accessor_dt.py", line 462, in weekday_name
    return self._date_field("weekday_name", object)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sam/Documents/GitHub/filebasedbim/venv/lib/python3.12/site-packages/xarray/core/accessor_dt.py", line 245, in _date_field
    result = _get_date_field(_index_or_data(self._obj), name, dtype)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sam/Documents/GitHub/filebasedbim/venv/lib/python3.12/site-packages/xarray/core/accessor_dt.py", line 139, in _get_date_field
    out = access_method(values, name)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sam/Documents/GitHub/filebasedbim/venv/lib/python3.12/site-packages/xarray/core/accessor_dt.py", line 97, in _access_through_series
    field_values = getattr(values_as_series.dt, name).values
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'DatetimeProperties' object has no attribute 'weekday_name'
```

### Anything else we need to know?

day_name() accepts a locale, and if not provided, defaults to 'en_US.utf8'.
Pandas also have the month_name() method which may be worth implementing at the same time.

Alternatively, if locale implementation is not desired, weekday_name should be depreciated and removed as Pandas > v1.5 is required.

### Environment

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.12.0 (main, Oct  5 2023, 15:52:37) [Clang 14.0.3 (clang-1403.0.22.14.1)]
python-bits: 64
OS: Darwin
OS-release: 22.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: None
LOCALE: (None, 'UTF-8')
libhdf5: None
libnetcdf: None

xarray: 2023.12.0
pandas: 2.1.4
numpy: 1.26.2
scipy: None
netCDF4: None
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: None
cftime: None
nc_time_axis: None
iris: None
bottleneck: None
dask: None
distributed: None
matplotlib: None
cartopy: None
seaborn: None
numbagg: None
fsspec: None
cupy: None
pint: None
sparse: None
flox: None
numpy_groupies: None
setuptools: 68.2.2
pip: 23.3.1
conda: None
pytest: 7.4.3
mypy: None
IPython: None
sphinx: None
</details>

### PR DESCRIPTION
**Pull request by nameloCmaS**
Expanding on #8610, I considered the below in deciding to remove the parameter `dt.weekday_name`:

1. The parameter has not been able to be used since v0.17.0 (#4942) which is when pandas v1.0.0 was set as the minimum required version. This is 3 years it has not worked.
2. No one has raised an issue re this in 3 years, so assume it is not used or required by the user base; thus it is unlikely to break any code.
3. It was not in the test suite.
4. There are more flexible ways to get date strings such as strftime() which account for localisation.

I considered a number of solutions with the above:

1. Add a warning and return an empty T_DataArray.
5. Return the weekday_name using strftime() or similar.
6. Remove the parameter and implement the pandas day_name method (but this isn't in cftime).

It should also be noted that the parameter does not work for both pandas.dt and cftime:

```
import xarray as xr
import dumpy as np
dates = xr.cftime_range(start="0001", periods=2, freq="MS", calendar="noleap")
da = xr.DataArray(dates)
da.dt.weekday_name
```

returns:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sam/miniconda3/envs/xarray-docs/lib/python3.10/site-packages/xarray/core/accessor_dt.py", line 462, in weekday_name
    return self._date_field("weekday_name", object)
  File "/Users/sam/miniconda3/envs/xarray-docs/lib/python3.10/site-packages/xarray/core/accessor_dt.py", line 245, in _date_field
    result = _get_date_field(_index_or_data(self._obj), name, dtype)
  File "/Users/sam/miniconda3/envs/xarray-docs/lib/python3.10/site-packages/xarray/core/accessor_dt.py", line 139, in _get_date_field
    out = access_method(values, name)
  File "/Users/sam/miniconda3/envs/xarray-docs/lib/python3.10/site-packages/xarray/core/accessor_dt.py", line 65, in _access_through_cftimeindex
    field_values = getattr(values_as_cftimeindex, name)
AttributeError: 'CFTimeIndex' object has no attribute 'weekday_name'
```

- [x] Closes #8610
- [x] Tests unaffected
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [x] Removed functions/methods from `api.rst` and `api-hidden.rst`

### CODE REVIEW

File: xarray/core/accessor_dt.py [hunk 1]
```diff
@@ -59,7 +59,8 @@ def _access_through_cftimeindex(values, name):
         field_values = _season_from_months(months)
     elif name == "date":
         raise AttributeError(
-            "'CFTimeIndex' object has no attribute `date`. Consider using the floor method instead, for instance: `.time.dt.floor('D')`."
+            "'CFTimeIndex' object has no attribute `date`. Consider using the floor method "
+            + "instead, for instance: `.time.dt.floor('D')`."
```
Comment (max-sixty) [inline xarray/core/accessor_dt.py:63]:
```suggestion
            "instead, for instance: `.time.dt.floor('D')`."
```

File: xarray/core/accessor_dt.py [hunk 2]
```diff
@@ -439,7 +440,7 @@ def weekofyear(self) -> DataArray:
 
         warnings.warn(
             "dt.weekofyear and dt.week have been deprecated. Please use "
-            "dt.isocalendar().week instead.",
+            + "dt.isocalendar().week instead.",
```
Comment (max-sixty) [inline xarray/core/accessor_dt.py:443]:
```suggestion
            "dt.isocalendar().week instead.",
```

Was this an auto-formatter?

It's not required in python; it's also a (vv small) performance drag

File: doc/whats-new.rst [hunk 3]
```diff
@@ -30,6 +30,8 @@ Breaking changes
 
 Deprecations
 ~~~~~~~~~~~~
+- The `dt.weekday_name` parameter is now depreciated and has been removed. (:issue:`8610`, :pull:`8664`)
```
Comment (max-sixty) [inline doc/whats-new.rst:33]:
```suggestion
- The `dt.weekday_name` parameter wasn't functional on modern pandas versions and has been removed. (:issue:`8610`, :pull:`8664`)
```

**Review by max-sixty — COMMENTED** [review]
> This is 3 years it has not worked.

OK, that's fairly persuasive!

---

Thanks @nameloCmaS !

File: xarray/core/accessor_dt.py (hunk 2 above)
Comment (nameloCmaS) [reply xarray/core/accessor_dt.py:443]: Not a linter, habit as PyLint flags it

**Pull request merged**